typescript/lib/parsers/openai.ts

```diff
@@ -700,17 +720,32 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
     const output = aiConfig.getLatestOutput(prompt);
     if (output != null) {
       if (output.output_type === "execute_result") {
-        // If the prompt has output saved, add it to the messages array
         if (output.metadata?.rawResponse?.role === "assistant") {
-          if (typeof output.data === "string") {
-            messages.push({
-              content: output.data,
-              // TODO (rossdanlm): Support function_call and don't rely on rawResponse
-              role: output.metadata?.rawResponse?.role,
-              function_call: output.metadata?.rawResponse?.function_call,
-              name: output.metadata?.rawResponse?.name,
-            });
+          output as ExecuteResult;
+          let content: string | null = null;
+          let function_call: FunctionData | undefined = undefined;
+
+          if (output.data?.hasOwnProperty("value")) {
+            const outputData = output.data as OutputDataWithValue;
+            if (typeof outputData.value === "string") {
+              content = outputData.value;
+            } else if (outputData.kind === "tool_calls") {
+              outputData as OutputDataWithToolCallsValue;
+              // Typescript schema does not support array of function calls yet,
+              // but Python does so doing this for forward compatibility
+              function_call =
+                outputData.value[outputData.value.length - 1].function;
+            }
+          } else if (typeof output.data === "string") {
+            content = output.data;
           }
```

Comment:
This can be simplified:

```
if (typeof output.data === "string") {
  content = output.data;
} else if (output.data.kind === "tool_calls") {
  // OutputDataWithToolCallsValue type should be automatically inferred by the TS compiler in this if block 
  const outputData: OutputDataWithToolCallsValue = output.data 
  if (Array.isArray(outputData)) { 
    functionCall = outputData.value[outputData.value.length - 1].function; 
  } else { 
    functionCall = outputData.value.function; 
  }
} else {
  // output.data.value should be inferred to be string in this block
  content = outputData.value;
}
```

Reply:
Btw I get this error when TS doesn't know the explicit type so I need to set it as something. Maybe I'm missing something but that's why I explicitly set the line `const outputData = output.data as OutputDataWithValue;`
<img width="610" alt="Screenshot 2023-12-27 at 14 48 54" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/04c9637d-d203-412a-bb6d-ebb279c2d52d">

Reply:
But yea I'll put the explicit null-check first to make it easier to read!